docs/up-to-date-check.md

```diff
@@ -36,11 +39,45 @@ allows custom logic to be executed when determining the set of items. The releva
 `Microsoft.Managed.DesignTime.targets` with names:
 
 - `CollectUpToDateCheckInputDesignTime`
-- `CollectUpToDateCheckOutputDesignTime`
 - `CollectUpToDateCheckBuiltDesignTime`
 
 Note that changes to inputs **must** result in changes to outputs. If this rule is not observed, then an input may
-have a timestamp after all outputs, which leads the up-to-date check to consider the project out-of-date after building.
+have a timestamp after all outputs, which leads the up-to-date check to consider the project out-of-date after building
+indefinitely. This can lead to longer build times.
+
+### Grouping inputs and outputs into sets
+
+For some advanced scenarios, it's necessary to partition inputs and outputs into groups and consider each separately.
+This can be achieved by adding `Set` metadata to the relevant items.
+
+For example, an ASP.NET project may use sets to group Razor `.cshtml` files with their output assembly `MyProject.Views.dll`,
+which is distinct from the other compilation target `MyProject.dll`. This could be achieved with something like:
+
+```xml
+<ItemGroup>
+  <UpToDateCheckInput Include="Home.csproj" Set="Views" />
```

Comment:
Was this meant to be Home.cshtml?

Reply:
Yes, thanks! I have much more muscle memory for csproj :)

```suggestion
  <UpToDateCheckInput Include="Home.cshtml" Set="Views" />
```